src/main/scala/strawman/collection/immutable/ListSet.scala

```diff
@@ -124,5 +127,10 @@ object ListSet extends IterableFactory[ListSet] {
 
   def empty[A]: ListSet[A] = EmptyListSet.asInstanceOf[ListSet[A]]
 
+  def newBuilder[A](): Builder[A, ListSet[A]] =
+    new ImmutableBuilder[A, ListSet[A]](empty) {
+      def add(elem: A): this.type = { elems = elems + elem; this }
```

Comment:
Any way to push this implementation up into the trait without losing performance?  There's a lot of repetition of this pattern.